barretenberg/cpp/src/barretenberg/client_ivc/client_ivc.cpp

```diff
@@ -358,6 +360,70 @@ void ClientIVC::complete_kernel_circuit_logic(ClientCircuit& circuit)
     }
 }
 
+HonkProof ClientIVC::construct_oink_proof(const std::shared_ptr<DeciderProvingKey>& proving_key,
+                                          const std::shared_ptr<MegaVerificationKey>& honk_vk,
+                                          const std::shared_ptr<Transcript>& transcript)
+{
+    vinfo("computing oink proof...");
+    MegaOinkProver oink_prover{ proving_key, honk_vk, transcript };
+    oink_prover.prove();
+
+    fold_output.accumulator = proving_key; // initialize the prover accum with the completed key
```

Comment:
I may try to get rid of or simplify this global state (`fold_output`) - that would allow this method to be static